frontend/pages/code/[id]/chat.tsx

```diff
@@ -0,0 +1,141 @@
+import { FC, RefObject, useEffect, useRef, useState } from 'react'
+
+import { useSession } from 'next-auth/react'
+import { useRouter } from 'next/router'
+import * as socketIO from 'socket.io-client'
+import { getChatHistory } from '@/services/collaboration-service-api'
+
+interface ICollaborator {
+    name: string
+}
```

Comment:
can move this to types folder as well, i believe theres a file with collaboration api, so can paste into it.